Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -642,14 +644,22 @@ open class _ComposerVC<ExtraData: ExtraDataTypes>: _ViewController,
         _ picker: UIImagePickerController,
         didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey: Any]
     ) {
-        guard let url = info[UIImagePickerController.InfoKey.imageURL] as? URL else { return }
-        
         do {
-            let attachment = try AnyAttachmentPayload(localFileURL: url, attachmentType: .image)
-            content.attachments.append(attachment)
+            if let imageURL = info[.imageURL] as? URL {
+                // User picked an image
+                let attachment = try AnyAttachmentPayload(localFileURL: imageURL, attachmentType: .image)
+                content.attachments.append(attachment)
+            } else if let videoURL = info[.mediaURL] as? URL {
+                // User picked a video
+                // TODO: after video attachments (`VideoAttachmentInjector`) is properly implemented,
+                // the `attachmentType` should be `video`
+                let attachment = try AnyAttachmentPayload(localFileURL: videoURL, attachmentType: .file)
+                content.attachments.append(attachment)
```

Comment:
Hm, that's the first step, did I get it right? 🤔

The full LLC support would include:
- `.video` attachment type
- `VideoAttachmentPayload` + an alias for attachment with video payload
- `videoAttachments` exposed on the message model

Without ⬆️ changes, the videos can only be distinguished from files by checking the `FileAttachmentPayload.file`.

 Are we gonna have a follow-up PR?

Reply:
@evsaev very good points... how much effort do you foresee the full support would be?